processors/build.gradle.kts

```diff
@@ -13,5 +13,13 @@ java {
 
 dependencies {
   api("io.opentelemetry:opentelemetry-sdk")
+  compileOnly("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi")
+
+  // For EventToSpanEventBridge
+  implementation("io.opentelemetry:opentelemetry-exporter-otlp-common")
+  implementation("com.fasterxml.jackson.core:jackson-core")
```

Comment:
Yeah its required to call `Marshaler#writeJsonTo(OutputStream)`

`opentelemetry-exporter-logging-otlp` has the same dependency here: https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/logging-otlp/build.gradle.kts#L19

Reply:
Ok, that's what I was guessing. Can we change this to `runtimeOnly()` then?

Reply:
Weird... I get a compilation error when I switch it to runtime only:
```
Execution failed for task ':processors:compileJava'.
> Compilation failed; see the compiler output below.
  /Users/jberg/code/open-telemetry/opentelemetry-java-contrib/processors/src/main/java/io/opentelemetry/contrib/eventbridge/EventToSpanEventBridge.java:110: error: cannot access JsonGenerator
          marshaler.writeJsonTo(baos);
```

But this is unexpected because I'm not any part of `jackson-core` directly in any of the code of this project.

I tracked it down to the method signatures:
- `Marshaler#writeJsonTo(OutputStream)`
- `Marshaler#writeJsonTo(JsonGenerator)`

Even though I'm calling `Marhslaer#writeJsonTo(OutputStream)`, the fact that there is an overload which accepts `JsonGenerator` causes the compiler to error if `JsonGenerator` isn't on the compile classpath.

Confirmed this by publishing a local copy of `opentelemetry-java` renaming the second method to `Marshaler#writeJsonToGenerator(JsonGenerator)`. With this rename, we can get away with a `runtimeOnly` dependency.

Unfortunately that means for the purposes of this PR we need the dependency to be `implementation`.

Reply:
Not a big deal to have implementation instead of runtimeOnly, but I did go ahead and open a PR to change the method signature. If merged, we could update this to runtimeOnly: https://github.com/open-telemetry/opentelemetry-java/pull/6896

Reply:
Cool thank you for figuring this out. Totally fine with it being `implementation` for now, but I know eventually we'll get CVEs on it, which is why I asked. Thanks also for doing that PR.